OsmAnd/src/net/osmand/plus/routing/CurrentStreetName.java

```diff
@@ -77,12 +83,78 @@ public static CurrentStreetName getCurrentName(@NonNull RoutingHelper routingHel
 			if (rs != null) {
 				streetName.text = getRouteSegmentStreetName(routingHelper, rs, false);
 				streetName.turnType = TurnType.valueOf(TurnType.C, false);
-				streetName.shieldObject = rs.getObject();
+				streetName.shields = RoadShield.create(rs.getObject());
 			}
 		}
 		if (streetName.turnType == null) {
 			streetName.showMarker = true;
 		}
 		return streetName;
 	}
+
+	public static class RoadShield {
+		private final RouteDataObject rdo;
+		private final String tag;
+		private final String value;
+		private StringBuilder additional;
```

Comment:
String is better?

Reply:
Further down the code we add more tags to the field "additional" in the loop. StringBuilder more efficient.